### ISSUE TITLE
ANyView?

### ISSUE THREAD

**Issue by kylebrowning** [Mijick/CalendarView#30]
Why is erasing to any view required? when I remove those erasures, the UI is no longer styled

**Comment by FulcrumOne** [comment]
Hey @kylebrowning,

I'll start by answering the second question - the reason is that the protocols expect the function to return `AnyView`, so if you put `any View` in there, the protocol will simply use the default implementation of the function as there is a mismatch of types.

And now moving on to a more difficult question for me - it's embarrassing to admit, but I don't remember why I did that - I probably didn't want to put too many typealias in there, but I'm not sure 😅

I'm planning to come back to this library in about 1.5 months (I'm currently working on [NavigationView](https://github.com/Mijick/NavigationView), then there will be [PopupView](https://github.com/Mijick/PopupView) and then I'm releasing MediaPicker), so I promise to come back to this question then 😅

Thanks for your question and have a nice day!

**Comment by kylebrowning** [comment]
Yeah unfortunately using AnyView breaks identity and is discouraged by Apple in most cases. :/.

**Comment by FulcrumOne** [comment]
yeah, I know this breaks the view hierarchy on re-rendering and if that's the only reason I did it (not sure tho), then it's an obvious mistake. Anyways, I promise to come back here ASAP 😉

**Comment by kylebrowning** [comment]
Thanks for the quick response.  LMK if you need help.
